tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/SdkBuildTasks/ArchiveSymbolsTask.cs

```diff
@@ -0,0 +1,522 @@
+        }
+
+        /// <summary>
+        /// List of symbol archive request files
+        /// This is primarily used for test purpose
+        /// </summary>
+        [Output]
+        public List<string> SymbolsRequestFileList { get; set; }
+
+
+        /// <summary>
+        /// In case the process of archiving symbols fail, this email will be used to communicate the current status of your archive request
+        /// </summary>
+        public string ArchiveSymbolsRequestStatusEmail { get; set; }
+
+        /// <summary>
+        /// This is an internal list that is used to process assembly list
+        /// Rather than dealing with ITaskItems, this is a list of ITaskItem.ItemSpec
+        /// </summary>
+        private List<string> BuiltAssemblyList
+        {
+            get
+            {
+                if(_assemblyList == null)
+                {
+                    if(BuiltAssemblyFileCollection != null)
+                    {
+                        if(BuiltAssemblyFileCollection.Any<ITaskItem>())
+                        {
+                            _assemblyList = BuiltAssemblyFileCollection.Select<ITaskItem, string>((item) => item.ItemSpec)?.ToList<string>();
+                        }
+                        else
+                        {
+                            _assemblyList = new List<string>();
+                        }
+                    }
+                }
+
+                return _assemblyList;
+            }
+        }
+
+        /// <summary>
+        /// In Jenkins ${BUILD_NUMBER} provides the JobId
+        /// </summary>
+        public string BuildJobId
+        {
+            get
+            {
+                if(string.IsNullOrWhiteSpace(_buildJobId))
+                {
+                    Random r = new Random(DateTime.Now.Minute);
+                    _buildJobId = string.Format("rnd{0}", r.Next(99999).ToString());
+                }
+
+                return _buildJobId;
+            }
+
+            set
+            {
+                _buildJobId = value;
+            }
+        }
+
+        /// <summary>
+        /// True: Will not execute to queue up symbols request, will only create the request file and copy symbols to the SysmbolArchiveRootDir
+        /// False: Will create request file as well as execute to queue up symbol archive request
+        /// </summary>
+        public bool SkipExecuteSymbolsRequest { get; set; }
+        
+        /// <summary>
+        /// Project name registered with Symbol Archiving service
+        /// If not provided default name will be used set by the Azure .NET SDK team
+        /// </summary>
+        public string ArchiveSymbolsRequestProjectName { get; set; }
+
+        public ArchiveSymbolsTask()
+        {
+            SymbolsRequestFileList = new List<string>();
+            SkipExecuteSymbolsRequest = false;
+        }
+
+        public override bool Execute()
+        {
+            TaskLogger.LogInfo("Executing Archiving Task.....");
```

Comment:
Why call TaskLogger directly here as opposed to just Log?

Reply:
I see you setup your own logging wrapper but why make it a static class as opposed to just using the built in logging for msbuild?

Reply:
@weshaggard currently TaskLogger is not setup to be a static class. Correct me if I am not understanding your comment accurately.
TaskLogger has bunch of helper wrappers to handle various scenarios/needs for our logging purpose but underneath it is using msbuild logger.
TaskLogger purpose is to provide few flavors of warpper APIs that will be helpful to log around various collections.
Plus I will need some form of common API to switch between commandline logging and logging during debugging/running tests with debug Traces during tests.
I am open for suggestions if you think their is a better way to achieve this.

Reply:
If it were me I would probably use the default msbuild Log property to do the logging and if there were some extra wrappers I wanted I would make them extension methods on top of that. That way you don't have to worry about creating your custom NetSdkTaskLogger class and managing that.

Reply:
@weshaggard I see what you mean.
But here is my concern with the proposed route.....
If I go the extension method route, I will have to start checking in my wrappers if the BuildManager is initialized, because my tests will start failing.
But that means, I cannot use the default/out of the box log API directly as I will have to also check again if BuildManager is initialized.
So thought of creating a wrapper for my log purpose, unless their is a better way that you can think of.

Reply:
In the past for testing or running outside of msbuild context I've generally created a dummy logger that I used to initialize the task with so that I don't ever have to worry about checking if the build manager is initialized.

Reply:
@weshaggard we can iterate over this and if there is a better way, I can integrate it later on.
I am considering this as good to have and will go ahead and merge this PR.
Thanks for the input.

Reply:
Yes this is a nice to have just some random feedback while reviewing the code.